tough/src/lib.rs

```diff
@@ -400,18 +477,23 @@ fn check_expired<T: Role>(datastore: &Datastore, role: &T) -> Result<()> {
     Ok(())
 }
 
-fn parse_url<S: AsRef<str>>(url: S) -> Result<Url> {
-    let mut url = Cow::from(url.as_ref());
-    if !url.ends_with('/') {
-        url.to_mut().push('/');
+/// Checks to see if the `Url` has a trailing slash and adds one if not. Without a trailing slash,
+/// the last component of a `Url` is considered to be a file. `metadata_url` and `targets_url`
+/// must refer to a base (i.e. directory), so we need them to end with a slash.
+fn parse_url(url: Url) -> Result<Url> {
+    let mut s = url.to_string();
+    if s.as_str().ends_with('/') {
+        Ok(url)
+    } else {
```

Comment:
```suggestion
    if url.as_str().ends_with('/') {
        Ok(url)
    } else {
        let mut s = url.to_string();
```
No need to build the string if we're not going to append.

Reply:
Fixed in [6ce9dc8](https://github.com/awslabs/tough/compare/231ceaa8956d1fadd85c58e7f8202daff603bd5b..6ce9dc89c1658d2a3a11c9b6ef847cfe10f8a3fd)